From 7770c5a8d65cff1c590d2adb3ae1fd25f955def2 Mon Sep 17 00:00:00 2001
From: Alexandre Bailon <abailon@baylibre.com>
Date: Mon, 8 Nov 2021 08:57:31 +0000
Subject: [PATCH] apu: Fix the logic off apu_device_online

In the case of error, apu_device_online returns a error code.
But an error usually means that the APU is offline (or doesn't exist)
so we must return 0 instead of the error code.

Signed-off-by: Alexandre Bailon <abailon@baylibre.com>
---
 apu/apu_drm.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/apu/apu_drm.c b/apu/apu_drm.c
index 00ea6b9c..9fbd861c 100644
--- a/apu/apu_drm.c
+++ b/apu/apu_drm.c
@@ -634,7 +634,7 @@ drm_public int apu_device_online(struct apu_drm_device *dev)
 
 	ret = drmCommandWriteRead(dev->fd, DRM_APU_STATE, &req, sizeof(req));
 	if (ret)
-		return ret;
+		return 0;
 
 	return req.flags & APU_ONLINE;
 }
-- 
2.25.1

